benches/benches/bevy_ecs/schedule.rs

```diff
@@ -0,0 +1,62 @@
+use bevy_app::App;
+use bevy_ecs::prelude::*;
+use criterion::{criterion_group, criterion_main, Criterion};
+
+criterion_group!(benches, build_schedule);
+criterion_main!(benches);
+
+fn build_schedule(criterion: &mut Criterion) {
+    // empty system
+    fn sys() {}
+
+    // Use multiple different kinds of label to ensure that dynamic dispatch
+    // doesn't somehow get optimized away.
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, SystemLabel)]
+    struct NumLabel(usize);
```

Comment:
I actually think this is likely to be more realistic (and more robust to the changes in #4957) if we use multiple distinct types via const generics here.

Reply:
Would this be possible? I can't think of any way to actually `initialize` all of those const generics. I would need some way to do a `const for` loop, and I don't know of any workarounds to achieve that.

As far as compatibility with #4957, it's not really an issue. For the sake of that, I just convert it to a string and then leak it (before starting the benchmark).

Reply:
Yeah, you'd need macros. Probably not worth it, since there's the string-leaking workaround.